mozilla-release/mobile/android/base/java/org/mozilla/gecko/util/HttpHandler.java

```diff
@@ -0,0 +1,70 @@
+package org.mozilla.gecko.util;
+
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Copyright © Cliqz 2018
+ */
+
+public class HttpHandler {
```

Comment:
Did you check, if mozilla already has some class/method which does this job?

Reply:
They have BaseAction.java that contain buildHttpURLConnection()
and Util.java that contain getJsonResponse();
but the problem is BaseAction is an abstract class and we can't do multiple extend while using class that extend Loader.